jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.cpp

```diff
@@ -103,9 +105,13 @@ ecma_builtin_date_prototype_to_string (ecma_value_t this_arg) /**< this argument
 {
   ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
 
-  if (ecma_object_get_class_name (ecma_get_object_from_value (this_arg)) != LIT_MAGIC_STRING_DATE_UL)
+  if (!ecma_is_value_object (this_arg))
   {
-    ret_value = ecma_raise_type_error ("Incomplete Date type");
+    ret_value = ecma_raise_type_error ("Incompatible type");
+  }
+  else if (ecma_object_get_class_name (ecma_get_object_from_value (this_arg)) != LIT_MAGIC_STRING_DATE_UL)
+  {
+    ret_value = ecma_raise_type_error ("Incompatible type");
```

Comment:
As actions under the conditions of the if and else if blocks are the same, and conditions are not complex, it is better to merge the blocks.